src/java/org/apache/cassandra/cql3/conditions/ColumnCondition.java

```diff
@@ -687,16 +957,21 @@ public static final class MultiCellUdtBound extends Bound
         /**
          * The conditions values.
          */
+        @Nonnull
         private final List<ByteBuffer> values;
 
         /**
          * The protocol version
          */
+        @Nonnull
         private final ProtocolVersion protocolVersion;
 
         public MultiCellUdtBound(ColumnMetadata column, Operator op, List<ByteBuffer> values, ProtocolVersion protocolVersion)
         {
             super(column, op);
+            checkNotNull(values);
+            checkNotNull(protocolVersion);
+            checkForUnsetValues(values);
             assert column.type.isMultiCell();
             this.values = values;
             this.protocolVersion = protocolVersion;
```

Comment:
nit: `rowValue()` doesn't need to be `final` on 986...already `private`

Reply:
Removed it